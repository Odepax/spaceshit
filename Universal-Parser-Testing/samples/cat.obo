cat {
	name : string -- in(2, 20)
	purr strength : float -- in(0.0, 1.0)
	is cute : bool -- default(true)
	is grumpy : bool -- default(false)
	----
	meow
} -- plural(:cats)
