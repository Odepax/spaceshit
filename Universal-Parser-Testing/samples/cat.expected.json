[
	{
		"type": {
			"name": "cat",
			"generics": []
		},
		"properties": [
			{
				"name": "name",
				"type": {
					"name": "string",
					"generics": []
				},
				"modifiers": [
					{
						"name": "in",
						"parameters": [
							{
								"type": "integer",
								"value": 2
							},
							{
								"type": "integer",
								"value": 20
							}
						]
					}
				]
			},
			{
				"name": "purr strength",
				"type": {
					"name": "float",
					"generics": []
				},
				"modifiers": [
					{
						"name": "in",
						"parameters": [
							{
								"type": "real",
								"value": 0.0
							},
							{
								"type": "real",
								"value": 1.0
							}
						]
					}
				]
			},
			{
				"name": "is cute",
				"type": {
					"name": "bool",
					"generics": []
				},
				"modifiers": [
					{
						"name": "default",
						"parameters": [
							{
								"type": "name",
								"value": "true"
							}
						]
					}
				]
			},
			{
				"name": "is grumpy",
				"type": {
					"name": "bool",
					"generics": []
				},
				"modifiers": [
					{
						"name": "default",
						"parameters": [
							{
								"type": "name",
								"value": "false"
							}
						]
					}
				]
			}
		],
		"operations": [
			{
				"name": "meow",
				"parameters": [],
				"type": {
					"name": "",
					"generics": []
				},
				"modifiers":  []
			}
		],
		"events": [],
		"modifiers": [
			{
				"name": "plural",
				"parameters": [
					{
						"type": "type",
						"value": {
							"name": "cats",
							"generics": []
						}
					}
				]
			}
		]
	}
]
